### PR TITLE
Bug ap 934/route returns incorrectly when collapse is used

### PR DESCRIPTION
**Pull request by mbasset**
Tracker Ticket: 
- https://quandl.atlassian.net/browse/AP-934
## Description
- Fix up styling based on rubocop definitions.
- Remove some pry debugging requires
- Utilize rspec-its project
- Fix edge case where collapse removes data when only `date` column is available.
- Minor version bump
## Note to the reviewer
- Small cleanup of code based on rubocop auto-clean.
- Only functional change is support for the `date` column.
## Is there a migration?
- No
## Note to QA
- There is a corresponding ticket in wikiposit that relies on this change.
## Dependency
- None

### CODE REVIEW

**Comment by ccleung** [comment]
can we please add the associated ticket to this?

**Comment by ccleung** [comment]
Please specify to QA which quandl projects need to be tested. Off the top of my head I think wikiposit, DataPI, possibly dynamodb depend on this? quandl minions...?

File: lib/quandl/operation/collapse.rb [hunk 1]
```diff
@@ -2,126 +2,130 @@
...
+
+        def valid_collapses
+          [:daily, :weekly, :monthly, :quarterly, :annual]
+        end
+
+        def collapse(data, frequency)
+          return data unless valid_collapse?(frequency)
+
+          # Special scenario where we are only fetching the `date` column
+          date_column_only = data.count > 0 && data[0].count == 1
```
Comment (ccleung) [inline lib/quandl/operation/collapse.rb]: is it safe to assume that the first column is always a date? Are there cases where we use quandl collapse and the first column is not a date? maybe @sergeiQ or @jbyck may be able to give some feedback

Comment (mbasset) [inline lib/quandl/operation/collapse.rb]: That's my assumption based on the existing code but it would be nice to confirm.

**Comment by mbasset** [comment]
@ccleung I think as long as only wikiposit updates to use this version of the gem anything else using it will stay on the old version and be ok. Those projects would need to `bundle update` to get this new version in which case the tests should cover any changes due to logic.

**Comment by mbasset** [comment]
@A-Scott-Rowe Can we add the labels to this project?

**Comment by ccleung** [comment]
:+1:

**Pull request merged**